src/rdkafka_transport.c

```diff
@@ -308,10 +308,14 @@ rd_kafka_transport_socket_recv0 (rd_kafka_transport_t *rktrans,
 
                 if (unlikely(r == SOCKET_ERROR)) {
 #ifdef _MSC_VER
-                        if (WSAGetLastError() == WSAEWOULDBLOCK)
+                        int errno_save = WSAGetLastError();
```

Comment:
How about consolidating the win and non-win code?
Something like this:
```
     int errno_save = socket_errno;
     if (errno_save == EAGAIN
#ifdef _MSC_VER
        || errno_save == WSAEWOULDBLOCK
#endif 
        )
          return sum;

      rd_snprintf(errstr, errstr_size, "%s", socket_strerror(errno_save));
      return -1;
```

Reply:
That's nice.